Gamescene revamp

Jonathan042 opened the pull request:
Note that on the very surface, basically nothing has changed aside from party fowl not changing the sprite of the special chicken across from it, and two special effects play by frame rather than at the same time (it's pretty tough to spot). If you want to see the special played in more than just a frame, changing SPECIALLENGTH in gamescene.cpp will do that, although there's button issues that can arise from buffers, and it doesn't show the first action done. SpecialChickenEffect was moved to gamescene to remedy the issue with spy chicken and because I realized that it would need to "pause" the game to show each effect. Overall, I finally "remastered" gamescene like I said I would at some point.

ZRich098 commented:
As a further point of clarification, some stack effects requires the opponent stack to resolve (i.e. swap opponent top and bottom chicken), which was why we had trouble pushing the code into Stack.cpp.  I believe the solution to this is to change the add() method in Stack to take in both a card and the opponent stack, and modify moose addToStack methods accordingly.

Jonathan042 commented:
The primary reason I moved the special chickens to gamescene is because it needed access to some fields in it that I couldn't figure out in stack, primarily skipState and cooldown which I added last night so that each special chicken would activate individually and look clean later on when we have animations. Spy chicken needing moose functions was just the icing on the cake. The AI is a bit of an issue though yeah, not sure how to work around that. I guess I could put it back in stack but I'd need to do something weird like return a tuple.
